src/ApiClient/SearchClient.php

```diff
@@ -27,22 +27,26 @@ public function query(
         DateTimeImmutable $starts = null,
         DateTimeImmutable $ends = null
     ) : PromiseInterface {
-        return $this->getRequest(
-            Uri::fromParts([
-                'path' => 'search',
-                'query' => build_query(['for' => $query] + array_filter([
-                    'page' => $page,
-                    'per-page' => $perPage,
-                    'sort' => $sort,
-                    'order' => $descendingOrder ? 'desc' : 'asc',
-                    'subject[]' => $subjects,
-                    'type[]' => $types,
-                    'use-date' => $useDate,
-                    'start-date' => $starts ? $starts->format('Y-m-d') : null,
-                    'end-date' => $ends ? $ends->format('Y-m-d') : null,
-                ])),
-            ]),
-            $headers
-        );
+        $uri = $this->createUri([
+            'path' => 'search',
+            'query' => [
+                'for' => $query,
+                'page' => $page,
+                'per-page' => $perPage,
+                'sort' => $sort,
+                'order' => $descendingOrder ? 'desc' : 'asc',
+                'subject[]' => $subjects,
+                'type[]' => $types,
+                'use-date' => $useDate,
+                'start-date' => $starts ? $starts->format('Y-m-d') : null,
+                'end-date' => $ends ? $ends->format('Y-m-d') : null,
+            ],
+        ]);
+
+        if (!isset(parse_query($uri->getQuery())['for'])) {
+            $uri = Uri::withQueryValue($uri, 'for', $query);
```

Comment:
A little annoying to change the order, but avoids having to `array_filter()` before `createUri()` everywhere. (Maybe there should be a `requiredQuery` input that doesn't get filtered?)